src/pass/lower_intrin.cc

```diff
@@ -136,8 +136,7 @@ class IntrinInjecter : public arith::IRMutatorWithAnalyzer {
 
     if (analyzer_->CanProveGreaterEqual(op->b, 0)) {
       // Common pass, positive divisor
-      if (analyzer_->CanProveGreaterEqual(op->a, 0) ||
-          analyzer_->CanProveGreaterEqual(e, 0)) {
```

Comment:
We want to keep the condition of e>=0, as it can sometimes imply a>=0 but we cannot directly prove it

Reply:
But `floordiv(a, b) = e >= 0` always holds at this point (because `b >= 0`, and floordiv is always non-negative in this case).

Reply:
not really, unless you meant floormod

Reply:
Sorry, I indeed meant floordmod (and we are in the FloorMod function).

Reply:
ah ok, sorry about that